File: Actividad 3/tarea3comentada.php
<!DOCTYPE html>
<!-- La declaración DOCTYPE le dice al navegador qué versión de HTML esperar -->
<html lang="es">
<head>
    <meta charset="UTF-8">
    <!-- El título del documento que aparecerá en la pestaña del navegador -->
    <title>Calculadora Simple</title>
    <!-- Vincula un archivo CSS externo que define el estilo de la página -->
    <link rel="stylesheet" href="style.css">
    <script>
        // Esta función valida los campos del formulario antes de enviarlos
        function validarFormulario() {
            // Obtiene los valores de los campos numéricos
            var num1 = document.forms["calculadora"]["num1"].value;
            var num2 = document.forms["calculadora"]["num2"].value;
            // Verifica si ambos campos están llenos
            if (num1 === "" || num2 === "") {
                alert("Ambos números son requeridos.");
                return false;
            }
            // Verifica si ambos campos son números válidos
            if (isNaN(num1) || isNaN(num2)) {
                alert("Por favor, ingrese solo números válidos.");
                return false;
            }
            // Si todo es correcto, permite que el formulario se envíe
            return true;
        }
    </script>
</head>
<body>
<div class="contenedor">
    <!-- Título de la página visible para el usuario -->
    <h1>Calculadora Simple</h1>
    <!-- El formulario que recoge los números y la operación deseada -->
    <form name="calculadora" action="<?php echo htmlspecialchars($_SERVER['PHP_SELF']); ?>" onsubmit="return validarFormulario()" method="post">
        <!-- Campos para ingresar los números. El atributo 'pattern' asegura que solo se ingresen números válidos -->
        <input type="text" pattern="-?[0-9]*(\.[0-9]+)?" title="Ingrese un número válido, positivo o negativo con o sin decimales." name="num1" required>
        <input type="text" pattern="-?[0-9]*(\.[0-9]+)?" title="Ingrese un número válido, positivo o negativo con o sin decimales." name="num2" required>

        <!-- Un menú desplegable para seleccionar la operación matemática -->
        <select name="operacion">
            <option value="suma">Suma</option>
            <option value="resta">Resta</option>
            <option value="multiplicacion">Multiplicación</option>
            <option value="division">División</option>
        </select>

        <!-- Botón para enviar el formulario -->
        <input type="submit" name="calcular" value="Calcular">
    </form>
</div>
<div id="resultado">
    <?php
    // Este bloque PHP se ejecuta cuando se envía el formulario
    if (isset($_POST['calcular'])) {
        // Filtra y valida los números ingresados como flotantes
        $num1 = filter_input(INPUT_POST, 'num1', FILTER_VALIDATE_FLOAT);
        $num2 = filter_input(INPUT_POST, 'num2', FILTER_VALIDATE_FLOAT);
        // Obtiene la operación seleccionada del formulario
        $operacion = $_POST['operacion'];

        // Verifica si los números son válidos
        if ($num1 === false || $num2 === false) {
            echo "Número(s) inválido(s).";
        } else {
            // Realiza la operación matemática seleccionada
            switch ($operacion) {
                case 'suma':
                    $resultado = $num1 + $num2;
                    break;
                case 'resta':
                    $resultado = $num1 - $num2;
                    break;
                case 'multiplicacion':
                    $resultado = $num1 * $num2;
                    break;
                case 'division':
                    // Verifica que no se divida por cero
                    if ($num2 != 0) {
                        $resultado = $num1 / $num2;
                    } else {
                        $resultado = "No se puede dividir por cero.";
                    }
                    break;
                default:
                    $resultado = "Operación no válida.";
                    break;
            }
            // Muestra el resultado de la operación
            echo "El resultado es: " . htmlspecialchars($resultado) ;
        }
    }
    ?>
</div>
</body>
</html>

File: Actividad 3/Tarea3comentada.c
#include <stdio.h>
#include <stdlib.h>

// Función para realizar la suma de dos números.
void opcion1() {
    int a, b, suma;
    char buffer[100]; // Buffer para manejar la entrada incorrecta.
    
    // Solicita el primer número hasta que sea válido.
    do {
        printf("Indica primer numero: ");
        if (scanf("%d", &a) != 1) {
            printf("Por favor, ingrese un numero valido.\n");
            scanf("%[^\n]", buffer); // Limpiar el búfer de entrada si no es un número.
        } else {
            break; // Salir del bucle si el número es válido.
        }
    } while (1);
    
    // Solicita el segundo número hasta que sea válido.
    do {
        printf("Indica segundo numero: ");
        if (scanf("%d", &b) != 1) {
            printf("Por favor, ingrese un numero valido.\n");
            scanf("%[^\n]", buffer); // Limpiar el búfer de entrada si no es un número.
        } else {
            break; // Salir del bucle si el número es válido.
        }
    } while (1);
    
    suma = a + b; // Realiza la suma.
    printf("La suma de %d + %d = %d\n", a, b, suma); // Muestra el resultado.
}

// Función para realizar la resta de dos números.
void opcion2() {
    // ... (Similar a la función opcion1 con la operación de resta)
}

// Función para realizar la multiplicación de dos números.
void opcion3() {
    // ... (Similar a la función opcion1 con la operación de multiplicación)
}

// Función para realizar la división de dos números.
void opcion4() {
    // ... (Similar a la función opcion1 con la operación de división y verificación de división por cero)
}

// Función principal que muestra el menú y maneja las opciones.
int main() {
    int op; // Variable para la opción seleccionada.
    int continuar = 1; // Controla la continuidad del menú.
    
    // Bucle principal del menú.
    while (continuar) {
        // Muestra las opciones del menú.
        printf("BIENVENIDO AL MENU\n");
        printf("1. Sumar\n");
        printf("2. Resta\n");
        printf("3. Multiplicacion\n");
        printf("4. Division\n");
        printf("0. Salir\n");
        
        // Solicita la opción del usuario.
        if (scanf("%d", &op) != 1) {
            printf("Por favor, ingrese un numero valido.\n");
            scanf("%*s"); // Limpiar el búfer de entrada si no es un número.
            continue; // Continúa al inicio del bucle si la entrada no es válida.
        }
        
        // Maneja la opción seleccionada.
        switch(op) {
            case 1:
                opcion1(); // Llama a la función de suma.
                break;
            case 2:
                opcion2(); // Llama a la función de resta.
                break;
            case 3:
                opcion3(); // Llama a la función de multiplicación.
                break;
            case 4:
                opcion4(); // Llama a la función de división.
                break;
            case 0:
                printf("Gracias por usar el programa\n"); // Mensaje de despedida.
                continuar = 0; // Establece 'continuar' en 0 para salir del bucle y terminar el programa.
                break;
            default:
                printf("Opcion no valida. Intente nuevamente.\n"); // Mensaje de error para opciones no reconocidas.
                break;
        }
    }
    
    return 0; // Retorna 0 para indicar la terminación exitosa del programa.
}
